libmamba/src/core/package_handling.cpp

```diff
@@ -84,12 +84,38 @@ namespace mamba
         return r;
     }
 
+    bool path_has_prefix(const fs::u8path& path, const fs::u8path& prefix)
+    {
+        auto pair = std::mismatch(path.std_path().begin(),
+                                  path.std_path().end(),
+                                  prefix.std_path().begin(),
+                                  prefix.std_path().end());
+        return pair.second == prefix.std_path().end();
+    }
+
+    int order(const fs::u8path& path)
+    {
+        int is_info = path_has_prefix(path, "info");
+        return !is_info;
+    }
+
+    int zip_order(const fs::u8path& path)
+    {
+        // sort info-...tar.zst file last in zip folder"
+        int init_order = starts_with(path.filename().string(), "info-");
+        // sort metadata.json first in zip folder
+        if (path.filename().string() == "metadata.json")
+            init_order = -1;
+        return init_order;
+    }
```

Comment:
info- is now last so that it could be buffered in memory while the pkg- archive is written directly to the zip